s3path/current_version.py

```diff
@@ -345,17 +346,17 @@ def rmdir(self):
             raise FileNotFoundError()
         accessor.rmdir(self)
 
-    def samefile(self, other_path: Union[str, S3Path]) -> bool:
+    def samefile(self, other_path: Union[str, os.PathLike[str]]) -> bool:
         """
         Returns whether this path points to the same Bucket key as other_path,
         Which can be either a Path object, or a string
         """
         self._absolute_path_validation()
-        if not isinstance(other_path, Path):
+        if not isinstance(other_path, S3Path):
```

Comment:
This is a behavior change, but I think it's actually a bugfix.

Reply:
I see
Nice one
good catch